core/controller/src/main/scala/org/apache/openwhisk/core/controller/actions/SequenceActions.scala

```diff
@@ -460,9 +465,10 @@ protected[actions] case class SequenceAccounting(atomicActionCnt: Int,
     // check conditions on payload that may lead to interrupting the execution of the sequence
     //     short-circuit the execution of the sequence iff the payload contains an error field
     //     and is the result of an action return, not the initial payload
-    val outputPayload = activation.response.result.map(_.asJsObject)
-    val payloadContent = outputPayload getOrElse JsObject.empty
-    val errorField = payloadContent.fields.get(ActivationResponse.ERROR_FIELD)
+    val errorField: Option[JsValue] = activation.response.result match {
+      case Some(JsObject(fields)) => fields.get(ActivationResponse.ERROR_FIELD)
+      case _                      => None
```

Comment:
So users can't define the explicit error response when the result is a JSON array.
But there would be no difference in the JSON-object-based action, right?

Sometimes, libraries return an `error` filed and OW considers that as an error of an action as well.
Since there would be no behavioral difference in the JSON-object-based actions, there would be no difference in existing semantics, right?

Reply:
* So users can't define the explicit error response when the result is a JSON array.
Yes, can't define the explicit error response for a JSON array,
but if we want to define the explicit error response, normally, user should return the JSON-object result.
* But there would be no difference in the JSON-object-based action, right?
For the JSON-object-based action, have no any bad influences on these actions, have no any change.
* Regarding
```
Sometimes, libraries return an error filed and OW considers that as an error of an action as well.
Since there would be no behavioral difference in the JSON-object-based actions, there would be no difference in existing semantics, right?
```
I am not sure whether i understand correctly, if libraries return an `error` or user's codes has some error, normally, there would has `catch(....)` statement in user's code, and in catch statement, user can return the error or some excetion error with JSON-object result.